.github/workflows/CI.yml

```diff
@@ -79,7 +79,22 @@ jobs:
       - name: Build and Run Tests
         run: |
           ./gradlew integTest yamlRestTest
-      - name: Multi Nodes Integration Testing
-        if: matrix.java == '21'
+  integMultiNodeTest:
+    needs: [spotless, javadoc]
```

Comment:
Nit: Should we also add `integTest` here? If `integTest` itself fails then we can avoid running multi node integ tests. WDYT?

Reply:
But our CI would run for a longer time then.

Reply:
Yeah I thought of that and figured we could run in parallel.